test/tests/functional/pbs_test_tpp.py

```diff
@@ -307,18 +377,90 @@ def test_mom_non_server_host(self):
         """
         msg = "Need atleast 2 moms which are running on a "
         msg += " non server host"
-        if len(self.moms) < 2 or self.server.shortname in self.moms.values():
+        if len(self.moms) < 2:
+            self.skip_test(reason=msg)
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        nodes = [self.hostA, self.hostB]
+        self.node_list.extend(nodes)
+        msg = "Requires mom which is running on non server host"
+        if self.server.shortname in nodes:
+            self.skip_test(reason=msg)
+        self.common_steps(job=True, resv=True,
+                          resv_job=True, client=self.hostA)
+        self.common_steps(job=True, interactive_job=True,
+                          client=self.hostB)
+
+    def test_comm_with_vnode_insertion(self):
+        """
+        Test for verifying vnode insertion when using TPP
+        """
+        if not self.mom.is_cpuset_mom():
+            a = {'resources_available.ncpus': 1}
+            self.server.create_vnodes('vn', a, 2, self.mom)
+            vnode_val = "vnode=vn[0]:ncpus=1+vnode=vn[1]:ncpus=1"
+        else:
+            vnode_val = "vnode=%s:ncpus=1" % self.server.status(NODE)[1]['id']
+            vnode_val += "+vnode=%s:ncpus=1" % self.server.status(NODE)[
+                2]['id']
+        set_attrib = {'Resource_List.select': vnode_val,
+                      ATTR_k: 'oe'}
+        self.common_steps(job=True, jset_attrib=set_attrib)
+        self.comm.stop('KILL')
+        if self.mom.is_cpuset_mom():
+            vnode_list = [self.server.status(NODE)[1]['id'],
+                          self.server.status(NODE)[2]['id']]
+        else:
+            vnode_list = ["vn[0]", "vn[1]"]
+        for vnode in vnode_list:
+            self.server.expect(
+                VNODE, {
+                    'state': 'state-unknown,down'}, id=vnode)
+
+    @requirements(num_moms=2, num_comms=1)
```

Comment:
The "num_comms" in requirement decorator doesn't count the comm on server host if comms is already passed in -p.

Reply:
@Mahalty - do you mean it does not "consider" the comm on server host?

Reply:
I guess this is clarified with offline discussion that right now in PTL, in case of multiple moms/comms, all the hostnames need to be passed in '-p' irrespective of whether it is server or non-server hostname.